time-series/time-series-api/src/main/java/com/powsybl/timeseries/RegularTimeSeriesIndex.java

```diff
@@ -163,7 +166,18 @@ public Duration getTimeStep() {
     }
 
     private static long computePointCount(Instant startTime, Instant endTime, Duration spacing) {
-        return Math.round(((double) (Duration.between(startTime, endTime).toNanos())) / spacing.toNanos()) + 1;
+        // Checks to avoid invalid duration and instants
+        if (startTime == null || endTime == null || spacing == null) {
+            throw new IllegalArgumentException("startTime, endTime, and spacing cannot be null.");
+        }
+
+        Duration duration = Duration.between(startTime, endTime);
+
+        if (duration.compareTo(MAX_DAYS) > 0 || spacing.compareTo(MAX_DAYS) > 0) {
+            throw new IllegalArgumentException("Time range or spacing exceeds " + MAX_DAYS + " days.");
+        }
+
+        return Math.round(((double) (duration.toNanos())) / spacing.toNanos()) + 1;
```

Comment:
Can you add the following unit test in `RegularTimeSeriesIndexTest`?

```java
    @Test
    void testContructorErrorMaxDaysReached() {
        Duration duration = Duration.ofDays(210 * 365);
        Instant i0 = Instant.ofEpochMilli(0);
        Instant i1 = Instant.ofEpochMilli(210L * 365 * 24 * 60 * 60 * 1000 + 1);
        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
                () -> new RegularTimeSeriesIndex(i0, i1, duration));
        assertEquals("Time range or spacing exceeds 73000 days.", exception.getMessage());
    }
```

I don't think we can test the other conditions with the existing code (errors are raised before calling `computePointCount(...)` in these cases).

Reply:
true.